extensions/standard-processors/controllers/UnorderedMapKeyValueStoreService.h

```diff
@@ -29,19 +29,21 @@
 #include "core/logging/Logger.h"
 #include "core/logging/LoggerConfiguration.h"
 #include "core/Resource.h"
+#include "controllers/keyvalue/PersistableKeyValueStoreService.h"
 
 namespace org {
 namespace apache {
 namespace nifi {
 namespace minifi {
 namespace controllers {
 
-class UnorderedMapKeyValueStoreService : virtual public KeyValueStoreService {
+/// Key-value store serice purely in RAM without disk usage
+class UnorderedMapKeyValueStoreService : virtual public PersistableKeyValueStoreService {
```

Comment:
If it's RAM only, then why is it `PersistableKeyValueStoreService`?

Reply:
Because AbstractCoreComponentStateManagerProvider (grandparent of this class) declares a pure virtual persistImpl() method that is implemented in base class PersistableKeyValueStoreService by calling persist(). Basically, persist() is a hook, that can be called with any implementation, doing a null operation when it is volatile.